Framework/HaptikLib.framework/Headers/Haptik.h

```diff
@@ -658,6 +684,36 @@ UIKIT_EXTERN NSNotificationName const HPOffersUpdatedNotification;
                controller:(__kindof UIViewController *)controller;
 
 
+/*!
+ Allows SDK clients to take user to a specific channel and trigger the Bot
+ @param viaName Represents the string key used to uniquely specify channel inside Haptik
+ @param message String message to be sent to the backend on opening channel screen
+ @param messageType Represents the messageType to be sent to the backend on opening channel screen
+ @param source Represents the source from where the specific channel is launched and the Bot is triggered
+ @param controller The current view controller over which the channel is expected to be presented
+ 
+ @code
+ 
+ #import "MyViewController.h"
+ 
+ - (void)openChannelInHaptikAndTriggerBot {
+ 
+ [[Haptik sharedSDK] launchChannelToTriggerBotFor:@"mychannelinhaptik"
+                                           message:@"MESSAGE_FOR_TRIGGERING_BOT"
+                                       messageType:@(MESSAGE_TYPE_HERE)
+                                            source:@"SOURCE_HERE"
+                                        controller:self];
+ }
+ 
+ @endcode
+ */
+- (void)launchChannelToTriggerBotFor:(NSString *)viaName
+                               message:(nullable NSString *)message
+                           messageType:(NSNumber *)messageType
```

Comment:
Android SDK has not exposed the `messageType` if this is supposed to be used by clients. In Android a default value is used I think.
Why this was done? Just to avoid confusion with clients regarding what value to pass for this parameter @iamsimranjot

Reply:
When I asked @imankur about it, we decided that `messageType` has to be sent by the clients. @tejas-haptik

Reply:
So that In future, if the backend needs any other type, it's scalable @tejas-haptik

Reply:
There's some confusion here. Apps send `message type = 47` if client invokes this method. @imankur @iamsimranjot

Reply:
@imankur Can you please check